Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -83,21 +83,26 @@ import BraintreeCore
         apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeStarted, isVaultRequest: shouldVault, linkType: linkType)
         let returnURLScheme = BTAppContextSwitcher.sharedInstance._returnURLScheme
 
-        if returnURLScheme.isEmpty {
-            NSLog(
-                "%@ Venmo requires a return URL scheme to be configured via [BTAppContextSwitcher setReturnURLScheme:]",
-                BTLogLevelDescription.string(for: .critical)
-            )
-            notifyFailure(with: BTVenmoError.appNotAvailable, completion: completion)
-            return
-        } else if let bundleIdentifier = bundle.bundleIdentifier, !returnURLScheme.hasPrefix(bundleIdentifier) {
-            NSLog(
-                // swiftlint:disable:next line_length
-                "%@ Venmo requires [BTAppContextSwitcher setReturnURLScheme:] to be configured to begin with your app's bundle ID (%@). Currently, it is set to (%@)",
-                BTLogLevelDescription.string(for: .critical),
-                bundleIdentifier,
-                returnURLScheme
-            )
+        if universalLink?.absoluteString.isEmpty == true || universalLink?.absoluteString == nil {
+            if returnURLScheme.isEmpty {
```

Comment:
Nit - these nested ifs are a little tricky to read. Can this instead just be combined into 1 if?

Reply:
Agree it's a little odd. We could do something like this, not sure if that's easier to read:
```swift
        let universalLinkEmpty: Bool = universalLink?.absoluteString.isEmpty == true || universalLink?.absoluteString == nil
        if universalLinkEmpty && returnURLScheme.isEmpty {
            NSLog(
                "%@ Venmo requires a return URL scheme or universal link to be configured.",
                BTLogLevelDescription.string(for: .critical)
            )
            notifyFailure(
                with: BTVenmoError.invalidReturnURL("Venmo requires a return URL scheme or universal link to be configured."),
                completion: completion
            )
            return
        } else if let bundleIdentifier = bundle.bundleIdentifier, !returnURLScheme.hasPrefix(bundleIdentifier) && universalLinkEmpty {
            NSLog(
                // swiftlint:disable:next line_length
                "%@ Venmo requires [BTAppContextSwitcher setReturnURLScheme:] to be configured to begin with your app's bundle ID (%@). Currently, it is set to (%@)",
                BTLogLevelDescription.string(for: .critical),
                bundleIdentifier,
                returnURLScheme
            )
        }
```
I thought it was a bit more clear to make it apparent that we only care about the return URL scheme/bundle ID if there is no universal link how it was written in the PR, but the above I think achieves the same.

Reply:
I've tested this approach:

```suggestion
        if universalLink?.absoluteString.isEmpty ?? true {
```

Reply:
@richherrera I am not sure I follow. Updating to this and commenting out `BTAppContextSwitcher.sharedInstance.returnURLScheme` in our `AppDelegate` returns an error with that update which is not expected.

Reply:
Sorry let me clarify: 

I was thinking that something like the following might work, instead of 
    `universalLink?.absoluteString.isEmpty == true || universalLink?.absoluteString == nil ` 
we can use 
     `universalLink?.absoluteString.isEmpty ?? true`

final approach suggested:

```swift
        if universalLink?.absoluteString.isEmpty ?? true, returnURLScheme.isEmpty {
            NSLog(
                "%@ Venmo requires a return URL scheme or universal link to be configured.",
                BTLogLevelDescription.string(for: .critical)
            )
            notifyFailure(
                with: BTVenmoError.invalidReturnURL("Venmo requires a return URL scheme or universal link to be configured."),
                completion: completion
            )
            return
        } else if let bundleIdentifier = bundle.bundleIdentifier, !returnURLScheme.hasPrefix(bundleIdentifier) {
            NSLog(
                "%@ Venmo requires [BTAppContextSwitcher setReturnURLScheme:] to be configured to begin with your app's bundle ID (%@). Currently, it is set to (%@)",
                BTLogLevelDescription.string(for: .critical),
                bundleIdentifier,
                returnURLScheme
            )
        }
```
I also removed `swiftlint:disable:next line_length` since it seems to be unnecessary.

Reply:
Ah gotcha, I think GitHub highlighted the wrong code. Yep we can do that if we feel like it's more clear. I normally add both the nil and empty checks because it makes it a bit more clear that `isEmpty` returns nil if you pass it a nil value. But if we feel like this pattern is explicit enough:
```swift
universalLink?.absoluteString.isEmpty ?? true
```
over this one:
```swift
universalLink?.absoluteString.isEmpty == true || universalLink?.absoluteString == nil 
```
I am okay with that. What do you think?

Reply:
Got it, I'm fine if it's possible to just combine both if statements

Reply:
Sounds good! Updated here: https://github.com/braintree/braintree_ios/pull/1503/commits/07780fc99cd1b08e1784545fde0cf5794a141f1a. Sorry it took me a bit wanted to test the flows again and kept forgetting to bring something to plug my phone into my laptop with to the office. 🤦